modules/images/webp-uploads/load.php

```diff
@@ -4,41 +4,290 @@
+
+	// Not a supported mime type to create the sources property.
+	$mime_type = get_post_mime_type( $attachment_id );
+	if ( ! isset( $valid_mime_transforms[ $mime_type ] ) ) {
+		return $metadata;
+	}
+
+	$file = get_attached_file( $attachment_id, true );
+
+	// File does not exist.
+	if ( ! file_exists( $file ) ) {
+		return $metadata;
+	}
+
+	$dirname     = pathinfo( $file, PATHINFO_DIRNAME );
+	$image_sizes = array();
+	if ( array_key_exists( 'sizes', $metadata ) && is_array( $metadata['sizes'] ) ) {
+		$image_sizes = $metadata['sizes'];
+	}
+
+	foreach ( wp_get_registered_image_subsizes() as $size_name => $properties ) {
+		// This image size does not exist on the defined sizes.
+		if ( ! isset( $image_sizes[ $size_name ] ) || ! is_array( $image_sizes[ $size_name ] ) ) {
+			continue;
+		}
+
+		$current_size = $image_sizes[ $size_name ];
+		$sources      = array();
+		if ( isset( $current_size['sources'] ) && is_array( $current_size['sources'] ) ) {
+			$sources = $current_size['sources'];
+		}
+
+		// Try to find the mime type of the image size.
+		$current_mime = '';
+		if ( isset( $current_size['mime-type'] ) ) {
+			$current_mime = $current_size['mime-type'];
+		} elseif ( isset( $current_size['file'] ) ) {
+			$current_mime = wp_check_filetype( $current_size['file'] )['type'];
+		}
+
+		if ( empty( $current_mime ) ) {
+			continue;
+		}
+
+		$sources[ $current_mime ] = array(
+			'file'     => array_key_exists( 'file', $current_size ) ? $current_size['file'] : '',
+			'filesize' => 0,
+		);
+
+		// Set the filesize from the current mime image.
+		$file_location = path_join( $dirname, $sources[ $current_mime ]['file'] );
+		if ( file_exists( $file_location ) ) {
+			$sources[ $current_mime ]['filesize'] = filesize( $file_location );
+		}
+
+		$formats = isset( $valid_mime_transforms[ $current_mime ] ) ? $valid_mime_transforms[ $current_mime ] : array();
+
+		foreach ( $formats as $mime ) {
+			wp_schedule_single_event( time(), 'webp_uploads_create_image', array( $attachment_id, $size_name, $mime ) );
+		}
+
+		$current_size['sources']         = $sources;
+		$metadata['sizes'][ $size_name ] = $current_size;
+	}
+
+	return $metadata;
+}
+
+add_filter( 'wp_generate_attachment_metadata', 'webp_uploads_create_sources_property', 10, 2 );
+
+/**
+ * Creates a new image based of the specified attachment with a defined mime type
+ * this image would be stored in the same place as the provided size name inside the
+ * metadata of the attachment.
+ *
+ * @since n.e.x.t
+ *
+ * @param int    $attachment_id The ID of the attachment we are going to use as a reference to create the image.
+ * @param string $size          The size name that would be used to create this image, out of the registered subsizes.
+ * @param string $mime          A mime type we are looking to use to create this image.
+ *
+ * @return bool|int|WP_Error
  */
-function webp_uploads_filter_image_editor_output_format( $output_format, $filename, $mime_type ) {
```

Comment:
rather than deleting the existing function, we may want to keep it around for sites that want to generate **only** WebP sub sized images (which is what the current code does)

Reply:
> rather than deleting the existing function, we may want to keep it around for sites that want to generate only WebP sub sized images (which is what the current code does)

This is a great point. Question how can we answer this one from the code?

> sites that want to generate only WebP sub sized images (which is what the current code does

Should we rely on the existing setting from the plugin for this scenario?

Reply:
@mitogh @adamsilverstein I think for now it's okay to remove the function since at this point we're focusing on the default implementation. However, we need to iterate on this in a follow-up issue (and later PR): We have to define a filter where developers can determine which formats they want to generate their images in. Once we introduce such a filter, we can reintroduce a filter for the output format, based on that filter's value.